src/features/Authentication/signup/signup-container.jsx

```diff
@@ -21,16 +22,20 @@ function SignUpContainer() {
   const {
     register,
     handleSubmit,
-    formState: { errors, isValid, isSubmitting },
+    formState: { errors, isValid, isSubmitting, trigger },
     reset,
   } = useForm({
     mode: "all",
     resolver: yupResolver(schema),
   });
 
-  const onSubmitHandler = (data) => {
-    console.log({ data });
-    reset();
+  const onSubmitHandler = async (data) => {
+    try {
+      await signup(data.email, data.password);
+      reset();
+    } catch (error) {
+      console.log(error);
+    }
   };
```

Comment:
How will the user know if there is an error, for example the email or password is wrong.
In the catch block where there is an error, firebase throws an error message to help you know what went wrong

What you can do is to set it to state and display it to the user


```jsx
const [errorMessage, setErrorMessage] = useState('')

try {
// -- signup code

} catch(error){
setErrorMessage(error.message)
}

```